services/src/main/java/io/druid/cli/CliHistorical.java

```diff
@@ -103,9 +114,52 @@ public void configure(Binder binder)
             JsonConfigProvider.bind(binder, "druid.historical.cache", CacheConfig.class);
             binder.install(new CacheModule());
             MetricsModule.register(binder, CacheMonitor.class);
+
+            binder.bind(ForSideEffectsOnlyProvider.Child.class).toProvider(ForSideEffectsOnlyProvider.class).asEagerSingleton();
           }
         },
         new LookupModule()
     );
   }
+
+  private static class ForSideEffectsOnlyProvider implements Provider<ForSideEffectsOnlyProvider.Child>
```

Comment:
It's kinda annoying that you have a provider in each of these CLIs.  The only deviation between these is the node type and the map.  You *should* be able to create a single provider that takes on the constructor a `String nodeType, List<Key<DruidService>> services`.  Move the injections to an 

```
@Inject
public void init(Injector injector, Lifecycle lifecycle, ...)
```

Then in the `get()` you would walk the list of `Key`s, asking the injector to inject them, build the map out of those and then use that to build the `discoveryDruidNode`.

Once that is setup, your module code would look like

```
binder.bind(ForSideEffectsOnlyProvider.Child.class).toProvider(new ForSideEffectsOnlyProvider("historical", Arrays.asList(Key.get(DataNodeService.class, LookupNodeService.class)));
LifecycleModule.register(ForSideEffectsOnlyProvider.Child.class);
```